RHDEVDOCS-6452: Content creation for GitOps 1.17 RN (#96515)
openshift/openshift-docs: modules/gitops-release-notes-1-17-0.adoc

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes-1-17-0.adoc

:_mod-docs-content-type: REFERENCE

[id="gitops-release-notes-1-17-0_{context}"]
= Release notes for {gitops-title} 1.17.0

{gitops-title} 1.17.0 is available on {OCP} 4.12, 4.14, 4.15, 4.16, 4.17, 4.18, and 4.19.

[id="errata-updates-1-17-0_{context}"]
== Errata updates

[id="RHEA-XXXX:NNNN-gitops-1-17-0-security-update-advisory_{context}"]

> === RHEA-XXXX:NNNN - {gitops-title} 1.17.0 security update advisory

Reviewer: I have just now noticed that there is this one more `===` level heading. I think this one could also use the similar format like the release notes items, so something like:

```
RHEA-XXXX:NNNN - {gitops-title} 1.17.0 security update advisory::
Issued: 2025-08-06
+
The list of enhancements that are included in this release are documented in the following advisory:
+
* link:https://access.redhat.com/errata/RHEA-XXXX:NNNN[RHEA-XXXX:NNNN]
```

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes-1-17-0.adoc

:_mod-docs-content-type: REFERENCE

[id="gitops-release-notes-1-17-0_{context}"]
= Release notes for {gitops-title} 1.17.0

{gitops-title} 1.17.0 is available on {OCP} 4.12, 4.14, 4.15, 4.16, 4.17, 4.18, and 4.19.

[id="errata-updates-1-17-0_{context}"]
== Errata updates

[id="RHEA-XXXX:NNNN-gitops-1-17-0-security-update-advisory_{context}"]

> === RHEA-XXXX:NNNN - {gitops-title} 1.17.0 security update advisory

Author: Yes, it is supposed to be updated when the additional errata details are available. I will leave this comment as unresolved till that time.

Issued: 2025-08-06

The list of enhancements that are included in this release are documented in the following advisory:

> * link:https://access.redhat.com/errata/RHEA-XXXX:NNNN[RHEA-XXXX:NNNN]

Author: Yes, it is supposed to be updated when the additional errata details are available. I will leave this comment unresolved till that time.

If you have installed the {gitops-title} Operator in the default namespace, run the following command to view the container images in this release:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator
----

[id="new-features-1-17-0_{context}"]
== New features

[id="gitops-1-17-0-extended-json-logging-support-for-additional-argo-cd-components_{context}"]
=== Extended JSON logging support to additional Argo CD components

* With this update, the `logFormat: json` setting in the Argo CD custom resource (CR) is supported by the `ApplicationSet` controller and `Notifications` controller. Previously, JSON logging was limited to components such as the API server, repo server, and application controller. link:https://issues.redhat.com/browse/GITOPS-6668[GITOPS-6668]
+
[NOTE]
====
Redis and Dex do not support configurable JSON logging and are unaffected by this change.

> ====

Author: Fixed.

[id="gitops-1-17-0-support-added-for-repeated-flags-in-extraCommandArgs_{context}"]
=== Support added for repeated flags in `extraCommandArgs`

* Before this update, the `extraCommandArgs` field in the Argo CD custom resource did not support using the same flag multiple times with different values. With this update, repeated flags are now correctly handled. Multiple instances of flags can be specified, and all values are included in the controller Pod configuration. link:https://issues.redhat.com/browse/GITOPS-6703[GITOPS-6703]

[id="gitops-1-17-0-fixed-UI-error-for-http-based-repository-credentials_{context}"]
=== Fixed UI error when editing HTTP-based repository credentials

> * Before this update, HTTP repository credentials could not be edited from the Argo CD Web UI, limiting the ability to manage HTTP-based repository configurations through the web interface. With this update, URL validation logic has been enhanced, and users can edit HTTP repository credentials directly in the Web UI. link:https://issues.redhat.com/browse/GITOPS-5969[GITOPS-5969]

Author: Fixed.

[id="deprecated-features-1-17-0_{context}"]
== Deprecated and removed features

[id="deprecation-of-some-fields-in-argo-cd_{context}"]
=== Deprecation of Keycloak support in OpenShift GitOps

> * In {gitops-title} 1.17, the support for Keycloak-based authentication is deprecated. When Keycloak is configured for single sign-on (SSO), deprecation warnings appear in the logs. Dex is the default and recommended authentication provider. It integrates with the OpenShift OAuth server and supports authentication using OpenShift users and groups. Users are required to migrate to Dex-based authentication for continued support. To continue using Keycloak, you must deploy and manage their own link:https://docs.redhat.com/en/documentation/red_hat_build_of_keycloak/26.0/html/operator_guide/installation-[Red Hat Build of Keycloak (RHBK)] instance and manually configure the integration in the Argo CD custom resource. link:https://issues.redhat.com/browse/GITOPS-7070[GITOPS-7070]

Author: Fixed.

[id="removed-support-for-deprecated-initialRepositories-repositoryCredentials-fields_{context}"]
=== Removed support for deprecated `initialRepositories` and `repositoryCredentials` fields

* In {gitops-title} 1.17, support for the deprecated `initialRepositories` and `repositoryCredentials` fields has been removed. If these fields are used, Argo CD logs a warning indicating their removal. Users can add repositories and repository credentials manually using the Argo CD CLI or the Argo CD dashboard. link:https://issues.redhat.com/browse/GITOPS-5961[GITOPS-5961]

[id="breaking-changes-1-17-0_{context}"]
== Breaking changes

[id="upgrade-to-argo-cd-3-0_{context}"]
=== Upgrade to Argo CD 3.0

> This release introduces an upgrade to the Argo CD component, transitioning from v2.14 to v3.0. This is a minor upgrade that introduces a list of breaking changes in Argo CD v3.0, with remediation steps provided in the following section. For more information, see link:https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.14-3.0/[Upgrading Argo CD from v2.0 to v3.0].

Author: Fixed.

[id="rbac-behavior-changes-with-argo-cd-dex-sso_{context}"]

> === RBAC behavior changes with Argo CD 3.0 and Dex SSO

Author: Fixed.

> * Argo CD 3.0 introduces a change in RBAC handling when using Dex for single sign-on (SSO). The RBAC system uses the `federated_claims.user_id` field instead of the previously encoded sub claim. This change affects how user identities are evaluated and matched in RBAC policies. Users must review and update their RBAC policies to ensure compatibility with Argo CD 3.0 when using the {gitops-title} Operator. For more information, see link:https://argocd-operator.readthedocs.io/en/latest/upgrading/#rbac-with-dex-sso-authentication[Configuring RBAC with Dex SSO Authentication].

Reviewer: Based on minimalism: I think removing ` when using the {gitops-title} Operator` does not really change the meaning, it seems assumed here. 
```suggestion
* Argo CD 3.0 introduces a change in RBAC handling when using Dex for single sign-on (SSO). The RBAC system uses the `federated_claims.user_id` field instead of the previously encoded sub claim. This change affects how user identities are evaluated and matched in RBAC policies. You must review and update your RBAC policies to ensure compatibility with Argo CD 3.0. For more information, see link:https://argocd-operator.readthedocs.io/en/latest/upgrading/#rbac-with-dex-sso-authentication[Configuring RBAC with Dex SSO Authentication].
```

+
--
link:https://issues.redhat.com/browse/GITOPS-6887[GITOPS-6887]
--

[id="added-guidance-for-logs-rbac-enforcement-in-argocd_{context}"]
=== Added guidance for logs RBAC enforcement in Argo CD 3.0

> * Before this update, users with access to Applications could also view logs.  With this update, Argo CD 3.0 treats logs as a distinct RBAC resource requiring explicit permissions. For more information, see link:https://argocd-operator.readthedocs.io/en/latest/upgrading/#logs-rbac-enforcement[Logs RBAC Enforcement]. link:https://issues.redhat.com/browse/GITOPS-6886[GITOPS-6886]

Reviewer: To remove anthropomorphism:
```suggestion
* Before this update, users with access to applications could also view logs.  With this update, Argo CD 3.0 uses a separate RBAC resource to control log access, which requires explicit permissions. For more information, see link:https://argocd-operator.readthedocs.io/en/latest/upgrading/#logs-rbac-enforcement[Logs RBAC Enforcement]. link:https://issues.redhat.com/browse/GITOPS-6886[GITOPS-6886]
```

Author: Fixed.

[id="gitops-1-17-0-fine-grained-rbac-inheritance-enabled-by-default_{context}"]
=== Fine-grained RBAC inheritance enabled by default

> * Starting with Argo CD v3.0, fine-grained RBAC inheritance of Applications has been disabled, making `update` and `delete` actions apply only to the application resource. To perform these actions on managed resources, you must configure new policies. To maintain the Argo CD v2.0 behavior, the `server.rbac.disableApplicationFineGrainedRBACInheritance` parameter is set to `false` by default in the `argocd-cm` config map. You can enable the new fine-grained RBAC functionality by setting the following configuration in the Argo CD custom resource:

Reviewer: ```suggestion
* Starting with Argo CD v3.0, fine-grained RBAC inheritance of applications is disabled, making `update` and `delete` actions apply only to the application resource. To perform these actions on managed resources, you must configure new policies. To maintain the Argo CD v2.0 behavior, the `server.rbac.disableApplicationFineGrainedRBACInheritance` parameter is set to `false` by default in the `argocd-cm` config map. You can enable the new fine-grained RBAC functionality by setting the following configuration in the Argo CD custom resource:
```

+
[source,YAML]
----
apiVersion: argoproj.io/v1beta1
kind: ArgoCD
metadata:
  name: example

> spec:
>   extraConfig:
>     server.rbac.disableApplicationFineGrainedRBACInheritance: true # <1>
> ----

Reviewer: Use ellipsis to make it clear that this is not complete YAML sample:
```suggestion
spec:
  # ...
  extraConfig:
    server.rbac.disableApplicationFineGrainedRBACInheritance: true # <1>
  # ...
----
```

Author: Fixed.

=== RBAC behavior changes with Argo CD 3.0 and Dex SSO

* Argo CD 3.0 introduces a change in RBAC handling when using Dex for single sign-on (SSO). The RBAC system uses the `federated_claims.user_id` field instead of the previously encoded sub claim. This change affects how user identities are evaluated and matched in RBAC policies.
+
To address this, the Argo CD Operator documentation has been updated with:

** Detailed migration guidance for transitioning from sub to federated_claims.user_id
** Step-by-step remediation procedures
** Practical RBAC policy examples
** Best practices for managing RBAC with the new hands-off approach
+
Users must review and update their RBAC policies to ensure compatibility with Argo CD 3.0 when using the {gitops-title} Operator. For more information, see link:https://argocd-operator.readthedocs.io/en/latest/upgrading/#rbac-with-dex-sso-authentication[Configuring RBAC with Dex SSO Authentication].
+
--
link:https://issues.redhat.com/browse/GITOPS-6887[GITOPS-6887]
--

[id="added-guidance-for-logs-rbac-enforcement-in-argocd_{context}"]
=== Added guidance for logs RBAC enforcement in Argo CD 3.0

> * Before this update, the Argo CD Operator upstream documentation did not explain the changes to logs RBAC introduced in Argo CD 3.0. In contrast to Argo CD 2.x, where users with access to Applications could also view logs, Argo CD 3.0 treats logs as a distinct RBAC resource requiring explicit permissions. With this update, the Argo CD Operator upstream documentation includes:

Author: Fixed.

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes-1-17-0.adoc

:_mod-docs-content-type: REFERENCE

[id="gitops-release-notes-1-17-0_{context}"]
= Release notes for {gitops-title} 1.17.0

{gitops-title} 1.17.0 is available on {OCP} 4.12 and later versions.

[id="errata-updates-1-17-0_{context}"]
== Errata updates

[id="RHEA-XXXX:NNNN-gitops-1-17-0-security-update-advisory_{context}"]
=== RHEA-XXXX:NNNN - {gitops-title} 1.17.0 security update advisory

Issued: 2025-08-06

The list of security fixes that are included in this release are documented in the following advisory:

* link:https://access.redhat.com/errata/RHEA-XXXX:NNNN[RHEA-XXXX:NNNN]

If you have installed the {gitops-title} Operator in the default namespace, run the following command to view the container images in this release:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator
----

[id="new-features-1-17-0_{context}"]
== New features

[id="gitops-1-17-0-argo-resource-tracking-method-updated-to-annotation-based_{context}"]
=== Resource tracking method updated to annotation-based

> * With this update, the default resource tracking method is `annotation-based` instead of `label-based`, aligning with the default tracking method introduced in Argo CD v3.0. link:https://issues.redhat.com/browse/GITOPS-6888[GITOPS-6888]

Author: Removed this as per the feedback received.

[id="gitops-1-17-0-fixed-rolebinding-errors-by-long-SourceNamespace-project-names_{context}"]
=== Fixed `RoleBinding` errors caused by long `SourceNamespace` project names

* Before this update, the Argo CD Operator failed to create `RoleBindings` when `sourceNamespaces` contained long namespace names. This issue was caused by Kubernetes label validation errors, as namespace names exceeding `63` characters exceeded the label value length limit. The Argo CD Operator attempted to use the full namespace name as a label value, resulting in reconciliation failures with errors such as `must be no more than 63 characters`. With this update, the issue has been fixed by introducing a `truncateWithHash` function that:

** Truncates names to comply with the 63-character limit
** Appends a deterministic hash suffix to maintain uniqueness
** Ensures consistent and unique label values for long namespace names
+
As a result, `RoleBindings` are now created successfully for long `sourceNamespaces`. link:https://issues.redhat.com/browse/GITOPS-7018[GITOPS-7018]

[id="gitops-1-17-0-fine-grained-rbac-inheritance-enabled-by-default_{context}"]
=== Fine-grained RBAC inheritance enabled by default

> * Before this update, fine-grained RBAC inheritance was disabled in Argo CD 3.0, restricting update and delete actions to the Application resources. This prevented users from modifying sub-resources without defining additional policies. With this update, fine-grained RBAC inheritance is enabled in the Argo CD config map by default. Users can control access to Application sub-resources using inherited RBAC rules in newer Argo CD versions. link:https://issues.redhat.com/browse/GITOPS-6889[GITOPS-6889]

Reviewer 2: If we are adding a new section `Major version upgrade of Argo CD 3.0` as suggested by Anand, we should move this RN there.

Also, the current phrasing makes it sound like a bug, but it's actually a behavioral change. Could you rephrase the text to align with below?
> Starting with Argo CD v3.0, the `update` and `delete` actions apply only to the `application` resource itself. To perform these actions on resources managed by the application, new policies must explicitly allow `update/*` or `delete/*` on those resources.
To preserve the Argo CD v2 behavior, the configuration value `server.rbac.disableApplicationFineGrainedRBACInheritance` is set to `false` by default in the Argo CD ConfigMap (`argocd-cm`).
Users can override this behavior to enable the new fine-grained RBAC functionality by setting the following in the ArgoCD custom resource:
```yaml 
apiVersion: argoproj.io/v1beta1
kind: ArgoCD
metadata:
  name: example
spec:
  extraConfig:
    server.rbac.disableApplicationFineGrainedRBACInheritance: true
```
> Refer to the [upstream documentation](https://argo-cd.readthedocs.io/en/stable/operator-manual/rbac/#fine-grained-permissions-for-updatedelete-action) for more details on the required policies.

Author: Fixed.

[id="gitops-1-17-0-extended-json-logging-support-for-additional-argo-cd-components_{context}"]
=== Extended JSON logging support to additional Argo CD components

* Before this update, setting `logFormat: json` in the Argo CD custom resource (CR) only enabled JSON logging for a limited set of components such as the API server, repo server, and application controller. With this update, setting `logFormat: json` is supported by the ApplicationSet controller and Notifications controller, allowing you to add logs in the JSON format. link:https://issues.redhat.com/browse/GITOPS-6668[GITOPS-6668]
+
[NOTE]
====
Redis and Dex do not support configurable JSON logging and are unaffected by this change.

> ====

Reviewer 2: Move this RN to features section.

[id="gitops-1-17-0-fixed-UI-error-for-http-based-repository-credentials_{context}"]
=== Fixed UI error when editing HTTP-based repository credentials

* Before this update, the Argo CD Web UI failed to handle repositories hosted over HTTP due to URL validation errors. This prevented users from editing credentials for HTTP-based repositories through the Web UI. With this update, URL validation logic has been enhanced, and users can edit HTTP repository credentials directly in the Web UI. link:https://issues.redhat.com/browse/GITOPS-5969[GITOPS-5969]

[id="gitops-1-17-0-authentication-tokens-redacted-in-argo-CD-e2e-test-logs_{context}"]
=== Authentication tokens redacted in Argo CD E2E test logs
 

> * Before this update, Argo CD end-to-end (E2E) tests exposed authentication tokens in logs due to missing command argument redaction. This posed a security risk by potentially compromising authentication tokens during test execution. With this update, an authentication token redactor has been integrated into E2E test command execution. Authentication tokens are now hidden in logs, enhancing test security and protecting user credentials. link:https://issues.redhat.com/browse/GITOPS-3942[GITOPS-3942]

Reviewer 3: AFAICT this can be removed, as it does not affect product code, only E2E tests (and our users/customers do not run E2E tests).

[id="removed-support-for-deprecated-initialRepositories-repositoryCredentials-fields_{context}"]
=== Removed support for deprecated `initialRepositories` and `repositoryCredentials` fields

* In {gitops-title} 1.17, support for the deprecated `initialRepositories` and `repositoryCredentials` fields has been removed. If these fields are used, Argo CD logs a warning indicating their removal. Users can add repositories and repository credentials manually using the Argo CD CLI or the Argo CD dashboard. link:https://issues.redhat.com/browse/GITOPS-5961[GITOPS-5961]

[id="breaking-change-1-17-0_{context}"]
== Breaking change

[id="rbac-behavior-changes-with-argo-cd-dex-sso_{context}"]

> === RBAC behavior changes with Argo CD 3.0 and Dex SSO

Reviewer 4: Can we add one section that says `Major version upgrade of Argo CD 3.0` and then add these breaking changes as sub sections. Also we can point to this upstream [link](https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.14-3.0/) for more details on the breaking changes and its mitigation. WDYT ?

Reviewer 2: +1

* Argo CD 3.0 introduces a change in RBAC handling when using Dex for single sign-on (SSO). The RBAC system uses the `federated_claims.user_id` field instead of the previously encoded sub claim. This change affects how user identities are evaluated and matched in RBAC policies.
+

> To address this, the Argo CD Operator documentation has been updated with:

Reviewer 2: Can you add the upstream operator documentation link somewhere in this text?
https://argocd-operator.readthedocs.io/en/latest/upgrading/#rbac-with-dex-sso-authentication

Author: Fixed.

+
[NOTE]
====
Redis and Dex do not support configurable JSON logging and are unaffected by this change.
====

[id="gitops-1-17-0-argocd-agent_{context}"]
=== Argo CD Agent

* With this update, the Argo CD Agent is available as a Technology Preview (TP) feature. A productized container image of the Argo CD Agent is available, along with a custom property in the Argo CD custom resource (CR) to enable the principal component in the hub cluster. For more information, see link:https://argocd-agent.readthedocs.io/latest/[Overview of the Argo CD Agent]. link:https://issues.redhat.com/browse/GITOPS-5016[GITOPS-5016]
+
[WARNING]
==== 	
The {gitops-shortname} `argocd-agent` feature is a Technology Preview feature.
====

[id="fixed-issues-1-17-0_{context}"]
== Fixed issues

[id="gitops-1-17-0-fixed-rolebinding-errors-by-long-SourceNamespace-project-names_{context}"]
=== Fixed `RoleBinding` errors caused by long `SourceNamespace` project names

> * Before this update, creating an OpenShift Route resource for the `ApplicationSet` webhook server failed when Argo CD instance names exceeded the Kubernetes label length limit of 63 characters or the hostname length limit of 253 characters. This caused validation errors and prevented the `ApplicationSet` webhook functionality from working. With this update, truncation logic shortens generated route names and hostnames while preserving uniqueness and complying with character limits. As a result, Argo CD instances with long names can now successfully create `ApplicationSet` webhook server routes without validation errors. link:https://issues.redhat.com/browse/GITOPS-7018[GITOPS-7018]

Reviewer 4: I thought this was fixed by Siddhesh's fix. Is it not ? If not let's remove this section and reopen the bug.

> * Before this update, creating an OpenShift Route resource for the `ApplicationSet` webhook server failed when Argo CD instance names exceeded the Kubernetes label length limit of 63 characters or the hostname length limit of 253 characters. This caused validation errors and prevented the `ApplicationSet` webhook functionality from working. With this update, truncation logic shortens generated route names and hostnames while preserving uniqueness and complying with character limits. As a result, Argo CD instances with long names can now successfully create `ApplicationSet` webhook server routes without validation errors. link:https://issues.redhat.com/browse/GITOPS-7018[GITOPS-7018]

Reviewer 5: Siddhesh's has fixed the appset route creation for regular ArgoCD instance. Instance with a longer name is still a bug as the secret creation with long name fails

> * Before this update, creating an OpenShift Route resource for the `ApplicationSet` webhook server failed when Argo CD instance names exceeded the Kubernetes label length limit of 63 characters or the hostname length limit of 253 characters. This caused validation errors and prevented the `ApplicationSet` webhook functionality from working. With this update, truncation logic shortens generated route names and hostnames while preserving uniqueness and complying with character limits. As a result, Argo CD instances with long names can now successfully create `ApplicationSet` webhook server routes without validation errors. link:https://issues.redhat.com/browse/GITOPS-7018[GITOPS-7018]

Author: Removing this issue from the content. Fixed.

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes-1-17-0.adoc

:_mod-docs-content-type: REFERENCE

[id="gitops-release-notes-1-17-0_{context}"]
= Release notes for {gitops-title} 1.17.0

{gitops-title} 1.17.0 is available on {OCP} 4.12 and later versions.

[id="errata-updates-1-17-0_{context}"]
== Errata updates

[id="RHEA-XXXX:NNNN-gitops-1-17-0-security-update-advisory_{context}"]
=== RHEA-XXXX:NNNN - {gitops-title} 1.17.0 security update advisory

Issued: 2025-08-06

The list of security fixes that are included in this release are documented in the following advisory:

* link:https://access.redhat.com/errata/RHEA-XXXX:NNNN[RHEA-XXXX:NNNN]

If you have installed the {gitops-title} Operator in the default namespace, run the following command to view the container images in this release:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator
----

[id="new-features-1-17-0_{context}"]
== New features

[id="gitops-1-17-0-argo-rollous-integration-ocp-dev-console-topology_{context}"]
=== Argo Rollouts integration in OpenShift Developer Console Topology view

> * With this update, a new Developer Preview feature adds support for displaying Argo Rollout resources in the OpenShift Developer Console's Topology view. Rollout instances now appear as individual nodes, improving visibility into progressive delivery workflows. To use this feature, install the GitOps 1.17.0 Operator on OpenShift Container Platform version 4.19.0 or later. link:https://issues.redhat.com/browse/GITOPS-5870[GITOPS-5870]

Reviewer 6: I still have to verify, but a required fix for a Console issue is in 4.19.6.  So, please put 4.19.6 or later here.

Also, what about the contents I added to the release note section of epic 5870 (Link above)?  I thought that was needed to warn users about the implications of this feature being a Developer Preview item.

Author: Hey @keithchong, let me get back to you on that soon.

[id="gitops-1-17-0-argo-resource-tracking-method-updated-to-annotation-based_{context}"]
=== Resource tracking method updated to annotation-based

* With this update, the default resource tracking method is `annotation-based` instead of `label-based`, aligning with the default tracking method introduced in Argo CD v3.0. link:https://issues.redhat.com/browse/GITOPS-6888[GITOPS-6888]

[id="fixed-issues-1-17-0_{context}"]
== Fixed issues

[id="gitops-1-17-0-fixed-rolebinding-errors-by-long-SourceNamespace-project-names_{context}"]

> === Fixed `RoleBinding` errors caused by long `SourceNamespace` project names

Reviewer 6: LGTM

* Before this update, the Argo CD Operator failed to create `RoleBindings` when `sourceNamespaces` contained long namespace names. This issue was caused by Kubernetes label validation errors, as namespace names exceeding `63` characters exceeded the label value length limit. The Argo CD Operator attempted to use the full namespace name as a label value, resulting in reconciliation failures with errors such as `must be no more than 63 characters`. With this update, the issue has been fixed by introducing a `truncateWithHash` function that:

** Truncates names to comply with the 63-character limit
** Appends a deterministic hash suffix to maintain uniqueness
** Ensures consistent and unique label values for long namespace names
+
As a result, `RoleBindings` are now created successfully for long `sourceNamespaces`. link:https://issues.redhat.com/browse/GITOPS-7018[GITOPS-7018]

[id="gitops-1-17-0-fine-grained-rbac-inheritance-enabled-by-default_{context}"]
=== Fine-grained RBAC inheritance enabled by default

> * Before this update, fine-grained RBAC inheritance was disabled in Argo CD 3.0, restricting update and delete actions to the Application resources. This prevented users from modifying sub-resources without defining additional policies. With this update, fine-grained RBAC inheritance is enabled in the Argo CD config map by default. Users can control access to Application sub-resources using inherited RBAC rules in newer Argo CD versions. link:https://issues.redhat.com/browse/GITOPS-6889[GITOPS-6889]

Reviewer 7: LGTM

[id="gitops-1-17-0-fixed-replica-count-for-redis-ha-haproxy-to-ensure-ha_{context}"]
=== Fixed replica count for `redis-ha-haproxy` to ensure high availability

* Before this update, a misconfiguration caused the `redis-ha-haproxy` deployment to use only one replica, resulting in intermittent access issues with the Redis cache. With this update, the number of `redis-ha-haproxy` replicas now matches the High Availability (HA) configuration. Two replicas are created by default, improving Redis cache availability and resilience. link:https://issues.redhat.com/browse/GITOPS-6806[GITOPS-6806]

[id="gitops-1-17-0-gitops-plugin-pods-comply-with-pod-security-policy_{context}"]
=== `gitops-plugin` Pods comply with Pod Security policy

> * Before this update, `gitops-plugin` Pods failed to start due to missing security context properties, resulting in non-compliance with the Pod security policy. This caused the {gitops-title} Operator to fail when creating these Pods. With this update, `gitops-plugin` includes the required security context settings, ensuring compliance with the Pod security policy. link:https://issues.redhat.com/browse/GITOPS-6777[GITOPS-6777]

Reviewer 6: The plugin pod failed to start instead of the GitOps operator failed, right @chetan-rns ?

Author: Fixed.

> * Before this update, `gitops-plugin` Pods failed to start due to missing security context properties, resulting in non-compliance with the Pod security policy. This caused the {gitops-title} Operator to fail when creating these Pods. With this update, `gitops-plugin` includes the required security context settings, ensuring compliance with the Pod security policy. link:https://issues.redhat.com/browse/GITOPS-6777[GITOPS-6777]

Reviewer 8: LGTM!

[id="gitops-1-17-0-fixed-invalid-route-hostnames-for-argo-cd-webhook-server_{context}"]
=== Fixed invalid Route hostnames for Argo CD webhook server

* Before this update, enabling `applicationSet.webhookServer.route` caused an invalid Route error due to the default hostname exceeding the Kubernetes label value limit. This issue resulted in application deployment failures. With this update, the default hostname is now generated using the Argo CD instance name and namespace, ensuring it stays within the valid length limits. This fix resolves the error and allows successful creation of the webhook server Route. link:https://issues.redhat.com/browse/GITOPS-6707[GITOPS-6707]

[id="gitops-1-17-0-support-added-for-repeated-flags-in-extraCommandArgs_{context}"]
=== Support added for repeated flags in `extraCommandArgs`

> * Before this update, the `extraCommandArgs` field in the Argo CD custom resource did not support using the same flag multiple times with different values. With this update, repeated flags are now correctly handled. Multiple instances of flags can be specified, and all values are included in the controller Pod configuration. link:https://issues.redhat.com/browse/GITOPS-6703[GITOPS-6703]

Reviewer 9: LGTM

[id="gitops-1-17-0-fixed-regular-for-repeated-flags-in-extraCommandArgs_{context}"]
=== Fixed regular expression handling in `sourceNamespaces`

* Before this update, incorrect regular expression processing in the `sourceNamespaces` field caused improper labeling of namespaces. As a result, the `argocd.argoproj.io/managed-by-cluster-argocd` label was not applied, affecting namespace management. With this update,, Argo CD correctly processes regular expressions in `sourceNamespaces`. The {gitops-title} Operator correctly applies labels based on regex patterns, enabling management of multiple namespaces using a single expression. link:https://issues.redhat.com/browse/GITOPS-6675[GITOPS-6675]

[id="gitops-1-17-0-extended-json-logging-support-for-additional-argo-cd-components_{context}"]
=== Extended JSON logging support to additional Argo CD components

> * Before this update, setting `logFormat: json` in the Argo CD custom resource (CR) only enabled JSON logging for a limited set of components such as the API server, repo server, and application controller. With this update, setting `logFormat: json` is supported by the ApplicationSet controller and Notifications controller, allowing you to add logs in the JSON format. link:https://issues.redhat.com/browse/GITOPS-6668[GITOPS-6668]

Reviewer 9: LGTM

+
[NOTE]
====
Redis and Dex do not support configurable JSON logging and are unaffected by this change.
====

[id="gitops-1-17-0-fixed-UI-error-for-http-based-repository-credentials_{context}"]

> === Fixed UI error when editing HTTP-based repository credentials

Reviewer 6: LGTM

* Before this update, the Argo CD Web UI failed to handle repositories hosted over HTTP due to URL validation errors. This prevented users from editing credentials for HTTP-based repositories through the Web UI. With this update, URL validation logic has been enhanced, and users can edit HTTP repository credentials directly in the Web UI. link:https://issues.redhat.com/browse/GITOPS-5969[GITOPS-5969]

[id="gitops-1-17-0-authentication-tokens-redacted-in-argo-CD-e2e-test-logs_{context}"]

> === Authentication tokens redacted in Argo CD E2E test logs

Reviewer 6: LGTM

[id="breaking-change-1-17-0_{context}"]
== Breaking change

[id="rbac-behavior-changes-with-argo-cd-dex-sso_{context}"]
=== RBAC behavior changes with Argo CD 3.0 and Dex SSO

* Argo CD 3.0 introduces a change in RBAC handling when using Dex for single sign-on (SSO). The RBAC system uses the `federated_claims.user_id` field instead of the previously encoded sub claim. This change affects how user identities are evaluated and matched in RBAC policies.
+
To address this, the Argo CD Operator documentation has been updated with:

** Detailed migration guidance for transitioning from sub to federated_claims.user_id
** Step-by-step remediation procedures
** Practical RBAC policy examples
** Best practices for managing RBAC with the new hands-off approach
+
Users must review and update their RBAC policies to ensure compatibility with Argo CD 3.0 when using the {gitops-title} Operator.
+
--
link:https://issues.redhat.com/browse/GITOPS-6887[GITOPS-6887]
--

[id="added-guidance-for-logs-rbac-enforcement-in-argocd_{context}"]
=== Added guidance for logs RBAC enforcement in Argo CD 3.0

> * Before this update, the Argo CD Operator upstream documentation did not explain the changes to logs access control introduced in Argo CD 3.0. In contrast to Argo CD 2.x, where users with access to Applications could also view logs, Argo CD 3.0 treats logs as a distinct RBAC resource requiring explicit permissions. With this update, the Argo CD Operator upstream documentation includes:

Reviewer 6: "...changes to logs access control...." to "...changes to logs RBAC..."  ? It's more specific.

[id="removed-support-for-deprecated-initialRepositories-repositoryCredentials-fields_{context}"]
=== Removed support for deprecated `initialRepositories` and `repositoryCredentials` fields

* In {gitops-title} 1.17, support for the deprecated `initialRepositories` and `repositoryCredentials` fields has been removed. If these fields are used, Argo CD logs a warning indicating their removal. Users can add repositories and repository credentials manually using the Argo CD CLI or the Argo CD dashboard. link:https://issues.redhat.com/browse/GITOPS-5961[GITOPS-5961]

[id="breaking-changes-1-17-0_{context}"]
== Breaking changes

[id="upgrade-to-argo-cd-3-0_{context}"]
=== Upgrade to Argo CD 3.0

> This release introduces an upgrade to the Argo CD component, transitioning from v2.0 to v3.0. This is a minor upgrade that contains only minor breaking changes. The following section includes a list of minor breaking changes affecting Argo CD v3.0 and some additional information to remediate the changes. For more information, see link:https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.14-3.0/[Upgrading Argo CD from v2.0 to v3.0].

Reviewer 4: We are transitioning from 2.14 to 3.0 
We can mention `for a complete list of breaking changes between 2.14 to 3.0, you can refer the following link`.

=== Fixed `RoleBinding` errors caused by long `SourceNamespace` project names

* Before this update, the Argo CD Operator failed to create `RoleBindings` when `sourceNamespaces` contained long namespace names. This issue was caused by Kubernetes label validation errors, as namespace names exceeding `63` characters exceeded the label value length limit. The Argo CD Operator attempted to use the full namespace name as a label value, resulting in reconciliation failures with errors such as `must be no more than 63 characters`. With this update, the issue has been fixed by introducing a `truncateWithHash` function that:

** Truncates names to comply with the 63-character limit
** Appends a deterministic hash suffix to maintain uniqueness
** Ensures consistent and unique label values for long namespace names
+
As a result, `RoleBindings` are now created successfully for long `sourceNamespaces`. link:https://issues.redhat.com/browse/GITOPS-7018[GITOPS-7018]

[id="gitops-1-17-0-fixed-replica-count-for-redis-ha-haproxy-to-ensure-ha_{context}"]
=== Fixed replica count for `redis-ha-haproxy` to ensure high availability

> * Before this update, a misconfiguration caused the `redis-ha-haproxy` deployment to use only one replica, resulting in intermittent access issues with the Redis cache. With this update, the number of `redis-ha-haproxy` replicas now matches the High Availability (HA) configuration. Two replicas are created by default, improving Redis cache availability and resilience. link:https://issues.redhat.com/browse/GITOPS-6806[GITOPS-6806]

Reviewer 5: @Rizwana777 please correct me if I am wrong

> * Before this update, a misconfiguration caused the `redis-ha-haproxy` deployment to use only one replica, resulting in intermittent access issues with the Redis cache. With this update, the number of `redis-ha-haproxy` replicas now matches the High Availability (HA) configuration. Two replicas are created by default, improving Redis cache availability and resilience. link:https://issues.redhat.com/browse/GITOPS-6806[GITOPS-6806]

Reviewer 9: yes, it should be 3

> * Before this update, a misconfiguration caused the `redis-ha-haproxy` deployment to use only one replica, resulting in intermittent access issues with the Redis cache. With this update, the number of `redis-ha-haproxy` replicas now matches the High Availability (HA) configuration. Two replicas are created by default, improving Redis cache availability and resilience. link:https://issues.redhat.com/browse/GITOPS-6806[GITOPS-6806]

Author: Fixed.

[id="gitops-1-17-0-gitops-plugin-pods-comply-with-pod-security-policy_{context}"]
=== `gitops-plugin` Pods comply with Pod Security policy

* Before this update, `gitops-plugin` Pods failed to start due to missing security context properties, resulting in non-compliance with the Pod security policy. With this update, `gitops-plugin` includes the required security context settings, ensuring compliance with the Pod security policy. link:https://issues.redhat.com/browse/GITOPS-6777[GITOPS-6777]

[id="gitops-1-17-0-fixed-invalid-route-hostnames-for-argo-cd-webhook-server_{context}"]
=== Fixed invalid Route hostnames for Argo CD webhook server

> * Before this update, enabling `applicationSet.webhookServer.route` caused an invalid Route error due to the default hostname exceeding the Kubernetes label value limit. This issue resulted in application deployment failures. With this update, the default hostname is now generated using the Argo CD instance name and namespace, ensuring it stays within the valid length limits. This fix resolves the error and allows successful creation of the webhook server Route. link:https://issues.redhat.com/browse/GITOPS-6707[GITOPS-6707]

Author: Fixed, please check @varshab1210 .

[id="gitops-1-17-0-support-added-for-repeated-flags-in-extraCommandArgs_{context}"]
=== Support added for repeated flags in `extraCommandArgs`

* Before this update, the `extraCommandArgs` field in the Argo CD custom resource did not support using the same flag multiple times with different values. With this update, repeated flags are now correctly handled. Multiple instances of flags can be specified, and all values are included in the controller Pod configuration. link:https://issues.redhat.com/browse/GITOPS-6703[GITOPS-6703]

[id="gitops-1-17-0-fixed-regular-for-repeated-flags-in-extraCommandArgs_{context}"]
=== Fixed regular expression handling in `sourceNamespaces`

> * Before this update, incorrect regular expression processing in the `sourceNamespaces` field caused improper labeling of namespaces. As a result, the `argocd.argoproj.io/managed-by-cluster-argocd` label was not applied, affecting namespace management. With this update, Argo CD correctly processes regular expressions in `sourceNamespaces`. The {gitops-title} Operator correctly applies labels based on regex patterns, enabling management of multiple namespaces using a single expression. link:https://issues.redhat.com/browse/GITOPS-6675[GITOPS-6675]

Reviewer 5: This has been added in 1.16.2 RN already

[id="gitops-1-17-0-fixed-UI-error-for-http-based-repository-credentials_{context}"]
=== Fixed UI error when editing HTTP-based repository credentials

> * Before this update, the Argo CD Web UI failed to handle repositories hosted over HTTP due to URL validation errors. This prevented users from editing credentials for HTTP-based repositories through the Web UI. With this update, URL validation logic has been enhanced, and users can edit HTTP repository credentials directly in the Web UI. link:https://issues.redhat.com/browse/GITOPS-5969[GITOPS-5969]

Reviewer 5: ```suggestion
* Before this update, the Argo CD Web UI failed to handle repositories hosted over HTTP due to URL validation errors. This prevented users from editing credentials for HTTP-based repositories through the Web UI. With this update, URL validation logic has been enhanced, and users can edit HTTP repository credentials directly in the Web UI. link:https://issues.redhat.com/browse/GITOPS-5969[GITOPS-5969]
```
```suggestion
* Before this update, HTTP repository credentials could not be edited from the Argo CD Web UI, limiting the ability to manage HTTP-based repository configurations through the web interface. With this update, URL validation logic has been enhanced, and users can edit HTTP repository credentials directly in the Web UI. link:https://issues.redhat.com/browse/GITOPS-5969[GITOPS-5969]
```

Author: Fixed.

[id="removed-support-for-deprecated-initialRepositories-repositoryCredentials-fields_{context}"]
=== Removed support for deprecated `initialRepositories` and `repositoryCredentials` fields

* In {gitops-title} 1.17, support for the deprecated `initialRepositories` and `repositoryCredentials` fields has been removed. If these fields are used, Argo CD logs a warning indicating their removal. Users can add repositories and repository credentials manually using the Argo CD CLI or the Argo CD dashboard. link:https://issues.redhat.com/browse/GITOPS-5961[GITOPS-5961]

> [id="breaking-changes-1-17-0_{context}"]

Reviewer 5: @anandf For breakng changes I think we should mention the actual updates instead of mentioning doc updates in upstream

Reviewer 4: @varshab1210 Do you mean the intialRepositories breaking change here ?

Reviewer 2: Let's incorporate latest suggestions made by @varshab1210

Reviewer 4: Yes, these are Actual breaking changes and not documentation changes. @Dhruv-Soni11 Can you correct it.

Author: Fixed.

=== RBAC behavior changes with Argo CD 3.0 and Dex SSO

* Argo CD 3.0 introduces a change in RBAC handling when using Dex for single sign-on (SSO). The RBAC system uses the `federated_claims.user_id` field instead of the previously encoded sub claim. This change affects how user identities are evaluated and matched in RBAC policies.
+
To address this, the Argo CD Operator documentation has been updated with:

** Detailed migration guidance for transitioning from sub to federated_claims.user_id
** Step-by-step remediation procedures
** Practical RBAC policy examples
** Best practices for managing RBAC with the new hands-off approach
+
Users must review and update their RBAC policies to ensure compatibility with Argo CD 3.0 when using the {gitops-title} Operator. For more information, see link:https://argocd-operator.readthedocs.io/en/latest/upgrading/#rbac-with-dex-sso-authentication[Configuring RBAC with Dex SSO Authentication].
+
--
link:https://issues.redhat.com/browse/GITOPS-6887[GITOPS-6887]
--

[id="added-guidance-for-logs-rbac-enforcement-in-argocd_{context}"]
=== Added guidance for logs RBAC enforcement in Argo CD 3.0

> * Before this update, the Argo CD Operator upstream documentation did not explain the changes to logs RBAC introduced in Argo CD 3.0. In contrast to Argo CD 2.x, where users with access to Applications could also view logs, Argo CD 3.0 treats logs as a distinct RBAC resource requiring explicit permissions. With this update, the Argo CD Operator upstream documentation includes:

Reviewer 5: Maybe refer to the upstream doc
https://argocd-operator.readthedocs.io/en/latest/upgrading/#logs-rbac-enforcement

cc: @anandf

** A detailed explanation of logs-specific RBAC enforcement
** Migration guides and remediation steps
** Examples for updating policies to include logs permissions
** Best practices for RBAC under the new hands-off management model
+
--
link:https://issues.redhat.com/browse/GITOPS-6886[GITOPS-6886]
--

[id="gitops-1-17-0-fine-grained-rbac-inheritance-enabled-by-default_{context}"]
=== Fine-grained RBAC inheritance enabled by default

> * Starting with Argo CD v3.0, the `update` and `delete` actions apply only to the application resource. To perform these actions on managed resources, you must configure policies to allow `update/\*` or `delete/\*`. To maintain the Argo CD v2.0 behavior, the `server.rbac.disableApplicationFineGrainedRBACInheritance` parameter is set to `false` by default in the `argocd-cm` config map. You can enable the new fine-grained RBAC functionality by setting the following configuration in the Argo CD custom resource:

Reviewer 5: ```suggestion
* Starting with Argo CD v3.0,  fine grained RBAC inheritance of Applications have been disabled making `update` and `delete` actions apply only to the application resource. To perform these actions on managed resources, you must configure new policies. To maintain the Argo CD v2.0 behavior, the `server.rbac.disableApplicationFineGrainedRBACInheritance` parameter is set to `false` by default in the `argocd-cm` config map to preserve the old RBAC behaviour. You can enable the new fine-grained RBAC functionality by setting the following configuration in the Argo CD custom resource:
```

Reviewer 5: cc: @anandf

Reviewer 5: @anandf should we mention upstream documentation as well?
https://argocd-operator.readthedocs.io/en/latest/reference/argocd/#fine-grained-rbac-for-application-update-and-delete-sub-resources-v30
